src/coreclr/jit/gentree.cpp

```diff
@@ -21308,6 +21308,158 @@ GenTree* Compiler::gtNewSimdCeilNode(var_types type, GenTree* op1, CorInfoType s
     return gtNewSimdHWIntrinsicNode(type, op1, intrinsic, simdBaseJitType, simdSize);
 }
 
+#if defined(TARGET_XARCH)
+GenTreeVecCon* Compiler::gtCvtCtrlTbl(var_types type, var_types sourceType, var_types targetType, unsigned simdSize)
```

Comment:
This looks to just be creating a broadcast node and is only used in one place...

Is there any reason we can't just inline it as many other uses do? Something like the following should be possible and a lot simpler:
```csharp
GenTreeVecCon* tblCon = gtNewVconNode(widenedType);
uint64_t iconVal = varTypeIsUnsigned(simdTargetBaseType) ? 0x08080088 : 0x00000088;

if (simdSourceBaseType == TYP_FLOAT)
{
    iconVal |= (iconVal << 32);
}

for (unsigned i = 0; i < (simdSize / 8); i++)
{
    tblCon->gtSimdVal.u64[i] = iconVal;
}
```

Reply:
It could potentially even just use `gtNewSimdCreateBroadcastNode` which is a pattern we use elsewhere for convenience, and which still gets optimized to a `GenTreeVecCon` node

Reply:
True. Made the suggested changes.